### PR TITLE
Add develop instructions to CONTRIBUTING

### PR DESCRIPTION
**Pull request by nicoddemus**
From: https://github.com/pytest-dev/pytest/pull/6244

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -262,6 +262,19 @@ Here is a simple overview, with pytest-specific bits:
 
    When committing, ``pre-commit`` will re-format the files if necessary.
 
+#. If instead of using ``tox`` you prefer to run the tests directly, then we suggest to create a virtual environment and use
+   an editable install with the ``testing`` extra::
+
+       $ python3 -m venv .env
```
Comment (blueyed) [inline CONTRIBUTING.rst]: s/.env/.venv/

`.env` is often used for envdirs etc, and `.venv` is more common from what I can tell.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -262,6 +262,19 @@ Here is a simple overview, with pytest-specific bits:
...
+       $ source .env/bin/activate  # linux
```
Comment (blueyed) [inline CONTRIBUTING.rst]: s/linux/Linux

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -262,6 +262,19 @@ Here is a simple overview, with pytest-specific bits:
...
+       $ .env/Scripts/activate.bat  # Windows
+       $ pip install -e .[testing]
```
Comment (blueyed) [inline CONTRIBUTING.rst]:
```suggestion
       $ pip install -e ".[testing]"
```

`[` might trigger globbing in shells, resulting in an error then.

**Comment by nicoddemus** [comment]
Cool, thanks for the review. 👍

**Comment by nicoddemus** [comment]
> If we recommend .venv/, it will be good to add it to .gitignore.

Yeah, now that you mention, that's why my gut feeling was using `.env` in the first place (that's on `.gitignore` already).

But no worries, added `.venv` to `.gitignore` as well.

**Pull request merged**